hazelcast/src/main/java/com/hazelcast/spi/impl/BasicOperationService.java

```diff
@@ -583,7 +643,12 @@ private void notifyRemoteCall(Response response) {
     /**
      * Responsible for processing an Operation.
      */
-    private final class OperationHandler implements BasicOperationHandler {
+    private final class BasicOperationHandler implements OperationHandler {
+
+        @Override
+        public Operation currentOperation() {
+            return null;
```

Comment:
Yes. This PR is not ready, but I want to give an indication of the direction we need to move to.